resources/validations/bin/validate_with_schematron.sh

```diff
@@ -1,69 +1,126 @@
+        # output directory root
+        -o)
+            shift
+            OUTPUT_ROOT="$1"
+            ;;
+        # Unknown option!
+        -*)
+            usage "Error: Unknown option: $1"
+            exit 1
+            ;;
+    esac
+    shift
+done
+
+echo output dir "${OUTPUT_ROOT}"
+echo to val "$DOC_TO_VALIDATE";
+if test ! -e  "$DOC_TO_VALIDATE" ; then
     echo "no file input for report, exiting"
     exit 1
+else 
+    echo "doc requested to be validated: ${DOC_TO_VALIDATE}"
 fi
-DOC_TO_VALIDATE="$1"
-echo "doc requested to be validated: ${DOC_TO_VALIDATE}"
 
 # Delete pre-existing XSLT report
 rm -rf target/*.xsl;
 
-SAXON_VERSION=$2
+#if version not specified default
 SAXON_VERSION=${SAXON_VERSION:-10.2}
 SAXON_OPTS="${SAXON_OPTS:-allow-foreign=true}"
 
 echo "using saxon version ${SAXON_VERSION}"
 
-mvn -q org.apache.maven.plugins:maven-dependency-plugin:2.1:get \
-    -DrepoUrl=https://mvnrepository.com/ \
-    -DartifactId=Saxon-HE \
-    -DgroupId=net.sf.saxon \
-    -Dversion="${SAXON_VERSION}"
+saxonLocation=saxon/Saxon-HE/"${SAXON_VERSION}"/Saxon-HE-"${SAXON_VERSION}".jar
+if test -n "$SAXON_CP" ; then
+    echo SAXON_CP env variable used is "${SAXON_CP}"
+elif command -v mvn &> /dev/null ;then
+    mvn -q org.apache.maven.plugins:maven-dependency-plugin:2.1:get \
+        -DrepoUrl=https://mvnrepository.com/ \
+        -DartifactId=Saxon-HE \
+        -DgroupId=net.sf.saxon \
+        -Dversion="${SAXON_VERSION}"
+    SAXON_CP=~/.m2/repository/net/sf/${saxonLocation}
+elif command -v curl &> /dev/null; then
+    SAXON_CP=lib/Saxon-HE-"${SAXON_VERSION}".jar
+    curl -H "Accept: application/zip" -o "${SAXON_CP}" https://repo1.maven.org/maven2/net/sf/"${saxonLocation}"
+else
+    echo "SAXON_CP environment variable is not set. mvn or curl is required to download dependencies, neither found, please install one and retry"
+    exit 1
+fi
 
 # Delete pre-existing SVRL report
-rm -rf report/schematron/*.results.xml
+rm -rf "${OUTPUT_ROOT}/report/schematron/*.results.xml"
+rm -rf "${OUTPUT_ROOT}/report/schematron/*.results.html"
 
-for qualifiedSchematronName in src/*.sch; do
+#in the future replace the for loop with an optional passed in directory or single schema file -f 
+for qualifiedSchematronName in "${SCHEMA_LOCATION_DIR}"/*.sch; do
     [ -e "${qualifiedSchematronName}" ] || continue
         
     # compute name without .sch
     schematronName=${qualifiedSchematronName##*/}
     schematronRoot=${schematronName%.*}
     
     # Use Saxon XSL transform to convert our Schematron to pure XSL 2.0 stylesheet
-    saxon_jar=~/.m2/repository/net/sf/saxon/Saxon-HE/"${SAXON_VERSION}"/Saxon-HE-"${SAXON_VERSION}".jar
 
-    java -cp "${saxon_jar}" net.sf.saxon.Transform \
+    java -cp "${SAXON_CP}" net.sf.saxon.Transform \
         -o:target/"${schematronRoot}".xsl \
         -s:"${qualifiedSchematronName}" \
         lib/schematron/trunk/schematron/code/iso_svrl_for_xslt2.xsl \
-        $SAXON_OPTS
+        "$SAXON_OPTS"
```

Comment:
Ironically, I removed quotes here because we want to be able to expand multiple arguments, even if currently have one. We declare it as a string with quotes at first in the variable binding to skip around building an array and having to splat it here, but allow expansion cleanly here.

I know shellcheck recommends against this in SC2086, but here I say it's a feature!

```suggestion
    # shellcheck disable=2086
    java -cp "${SAXON_CP}" net.sf.saxon.Transform \
        -o:target/"${schematronRoot}".xsl \
        -s:"${qualifiedSchematronName}" \
        lib/schematron/trunk/schematron/code/iso_svrl_for_xslt2.xsl \
        $SAXON_OPTS
```